server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -658,10 +797,17 @@ public void sendSearchResponse(InternalSearchResponse internalSearchResponse, At
             }
             listener.onResponse(buildSearchResponse(internalSearchResponse, failures, scrollId, searchContextId));
         }
+        onPhaseEnd(this);
+        setCurrentPhase(null);
```

Comment:
Lets set "none" phase instead of null to avoid null pointer exception. Though it will never ideally.